src/main/scala/strawman/collection/Seq.scala

```diff
@@ -18,20 +18,41 @@ trait LinearSeq[+A] extends Seq[A] with LinearSeqOps[A, LinearSeq, LinearSeq[A]]
 /** Base trait for indexed sequences that have efficient `apply` and `length` */
 trait IndexedSeq[+A] extends Seq[A] with IndexedSeqOps[A, IndexedSeq, IndexedSeq[A]]
 
+object IndexedSeq extends IterableFactory.Delegate[IndexedSeq](immutable.Vector)
+
 /** Base trait for Seq operations */
 trait SeqOps[+A, +CC[X], +C] extends Any
   with IterableOps[A, CC, C]
   with ArrayLike[A]
   with Equals {
 
+  /** Evidence that type `C` is (or can be converted to) an `IterableOnce[A]` */
+  implicit protected[this] def toCollection: C => IterableOnce[A]
```

Comment:
In order to provide a *default implementation* to some methods of `Seq` (e.g. `reverseIterator`), I had to add this implicit conversion from `C` to `IterableOnce[A]`. Note that putting an upper bound `C <: IterableOnce[A]` would not work here because `StringOps` and `ArrayOps` instantiate their `C` type to `java.lang.String` and `scala.Array[A]`, respectively, which would not conform to the upper bound.